test/controllers/user.test.js

```diff
@@ -316,4 +317,249 @@ describe('UserController', () => {
+          err.response.body.validationErrors.should.have.property('body');
+          err.response.body.validationErrors.body.should.be.an('array');
+          err.response.body.validationErrors.body.should.deep.include.members([
+            {
+              keyword: 'minimum',
+              dataPath: '.page_size',
+              schemaPath: '#/properties/page_size/minimum',
+              params: {
+                comparison: '>=',
+                limit: 1,
+                exclusive: false
+              },
+              message: 'should be >= 1'
+            }
+          ]);
+          done();
+        });
+    });
+
+    it('Should retrieve the first users page list', done => {
+      createUserAndSignIn()
+        .then(json => {
+          return request.get('/users').set(json.header, json.token);
+        })
+        .then(res => {
+          res.should.have.status(200);
+          res.should.be.json;
+          res.body.should.have.property('page_number');
+          res.body.page_number.should.be.a('number');
+          res.body.should.have.property('users');
+          res.body.users.should.all.be.an('array');
+          res.body.users.should.have.lengthOf(1);
+          res.body.should.have.property('pages_left');
+          res.body.pages_left.should.be.a('number');
+          res.body.should.have.property('overflow');
+          res.body.overflow.should.be.a('boolean');
+          done();
+        })
+        .catch(err => {
+          done(new Error(`User page not retrieved: ${err.message}`));
+        });
+    });
+
+    it('Should retrieve a users page list that does not exists', done => {
+      createUserAndSignIn()
+        .then(json => {
+          return request
+            .get('/users')
+            .set(json.header, json.token)
+            .send({
+              page_number: 2,
+              page_size: 10
+            });
+        })
+        .then(res => {
+          res.should.have.status(200);
+          res.should.be.json;
+          res.body.should.have.property('page_number');
+          res.body.page_number.should.be.a('number');
+          res.body.page_number.should.equal(2);
+          res.body.should.have.property('users');
+          res.body.users.should.all.be.an('array');
+          res.body.users.should.be.empty;
+          res.body.should.have.property('pages_left');
+          res.body.pages_left.should.be.a('number');
+          res.body.pages_left.should.be.equal(0);
+          res.body.should.have.property('overflow');
+          res.body.overflow.should.be.a('boolean');
+          res.body.overflow.should.equal(true);
+          done();
+        })
+        .catch(err => {
+          done(new Error(`User not be authenticated: ${err.message}`));
+        });
+    });
+
+    it('Should retrieve a users page list that has one page ahead', done => {
+      const tenPromisesUserCreation = [];
+      for (let index = 0; index < 10; index++) {
```

Comment:
would you abstract this to another place? you are free to call it how you like. we have a script for data creation, that could be a good place